comp/metadata/clusteragent/fx/fx.go

```diff
@@ -0,0 +1,23 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+// Package fx provides the fx module for the cluster-agent metadata component
+package fx
+
+import (
+	clusteragent "github.com/DataDog/datadog-agent/comp/metadata/clusteragent/def"
+	clusteragentimpl "github.com/DataDog/datadog-agent/comp/metadata/clusteragent/impl"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// Module defines the fx options for this component
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fxutil.ProvideComponentConstructor(
+			clusteragentimpl.NewComponent,
+		),
+		fxutil.ProvideOptional[clusteragent.Component](),
```

Comment:
Regarding my other comment, you would endup with a comp with no method. FX only instantiate something if something is needing it. This can be cumbersome in some use case as it forces you to update each main where you want this comp to be used.

A solution for this is to simply to always force the inclusion of you comp like [this](https://github.com/DataDog/datadog-agent/blob/main/comp/core/configsync/configsyncimpl/module.go#L50).